cli/azd/pkg/environment/manager.go

```diff
@@ -168,6 +179,28 @@ func (m *manager) LoadOrCreateInteractive(ctx context.Context, environmentName s
 				if !shouldCreate {
 					return nil, false, fmt.Errorf("environment '%s' not found: %w", environmentName, err)
 				}
+
+				// get subscription
+				subscriptionId, err := m.remoteEnv.PromptSubscription(ctx, "Select an Azure Subscription to use:")
+				if err != nil {
+					return nil, false, err
+				}
+				// get location
+				location, err := m.remoteEnv.PromptLocation(
+					ctx,
+					subscriptionId,
+					"Select an Azure location to use:",
+					m.locationFiler,
+				)
+				if err != nil {
+					return nil, false, err
+				}
+				// return env without persisting to local store
+				return NewWithValues(environmentName, map[string]string{
```

Comment:
With the introduction of multiple remote environment providers we cannot always know the user will know this data.  We may need to integrate with the provision provider here to retrieve the provision state.

Reply:
as I previously mentioned, 

> 2nd approach working on fixing the function that does this Environment.Manager.LoadOrCreateInteractive() ==> (some progress - using provisioing.Manager.EnsureSubscriptionAndLocation() creates circular dependency between packages)

It will create circular dependency.

Reply:
I'm not recommending calling the provision provider from the environment manager, but rather from the command that needs to ensure the environment is initialized.  The `azd env refresh` will only work on environments that have been previously provisioned.  So within the `cmd/env.go` we should be able to call `provisionManager.Initialize()` to initialize all the requirements for the provider - in this case subscription & location.

Try that out and let us know.

Reply:
on it

Reply:
@wbreza your concern was correct, I have tried it and it didn't work.

I think I have already highlighted this in my initial description for the pull request.
> 1st approach to remove the packages that have *environment.Environment found three projectManager, provisioningManager, env. ==> (Bad path - can't remove them as they are essential for the current functionality)

TBH, I want to close this pull request and take all the things that we have discussed and leave them as a comment on the original issue so, If someone works on it in the future they benefit from this.

Reply:
@john0isaac We really appreciate your contribution on the issue and others here! From reading the issue again, it does seem like whatever changes we would come up with would be feature-like, and may require a little more thought given the many new changes made to azd's core logic. The team is pretty busy with ongoing feature work recently, so I think it's okay for us to revisit this in the near future.

Reply:
Thanks @weikanglim, I will summarize everything and leave it in a comment on the GitHub issue.